sound/soc/sof/topology.c

```diff
@@ -3474,12 +3481,19 @@ EXPORT_SYMBOL(snd_sof_init_topology);
 
 int snd_sof_load_topology(struct snd_sof_dev *sdev, const char *file)
 {
+	struct firmware vfe_fw;
 	const struct firmware *fw;
 	int ret;
 
 	dev_dbg(sdev->dev, "loading topology:%s\n", file);
 
-	ret = request_firmware(&fw, file, sdev->dev);
+	if (sdev->pdata->vfe) {
+		fw = &vfe_fw;
+		ret = sof_ops(sdev)->request_topology(sdev, file, &vfe_fw);
```

Comment:
this looks really odd. I don't know how the guest would know what topology to use. that's for the host to provide.

Reply:
@plbossart as explained above, the idea is to use guest topology file names as guest identification. So, guest with ID "vm0" would obtain topology file "vm0.tplg".

Reply:
that provides no security at all. It's scary really that we would even contemplate the idea.

Reply:
@plbossart what exactly do you mean? That random guests shouldn't be allowed to get whatever topology they request? What dangers do you see, that any guest can obtain information about random audio pipelines? I understand, that in principle guests should only get information they're supposed to get, but what are the possible implications of this? How would you propose to solve this?